src/Symfony/Component/String/Tests/AbstractAsciiTestCase.php

```diff
@@ -729,4 +804,49 @@ public static function provideReplaceIgnoreCase()
             ['heMMo worMd', 3, 'hello world', 'L', 'M'],
         ];
     }
+
+    /**
+     * @dataProvider provideCamel
+     */
+    public function testCamel(string $expectedString, string $origin)
+    {
+        $instance = static::createFromString($origin)->camel();
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideCamel()
+    {
+        return [
+            ['', ''],
+            ['symfonyIsGreat', 'symfony_is_great'],
+            ['symfony5IsGreat', 'symfony_5_is_great'],
+            ['symfonyIsGreat', 'Symfony is great'],
+            ['symfonyIsAGreatFramework', 'Symfony is a great framework'],
+            ['symfonyIsGREAT', '*Symfony* is GREAT!!'],
+        ];
+    }
+
+    /**
+     * @dataProvider provideSnake
+     */
+    public function testSnake(string $expectedString, string $origin)
+    {
+        $instance = static::createFromString($origin)->snake();
+
+        $this->assertEquals(static::createFromString($expectedString), $instance);
+    }
+
+    public static function provideSnake()
+    {
+        return [
+            ['', ''],
+            ['symfony_is_great', 'symfonyIsGreat'],
+            ['symfony5_is_great', 'symfony5IsGreat'],
```

Comment:
@nicolas-grekas What is the expected result, `symfony5_is_great` or `symfony_5_is_great`?

Reply:
`symfony5_is_great`